src/qp_klp/StandardMetagenomicWorkflow.py

```diff
@@ -49,3 +60,141 @@ def __init__(self, **kwargs):
                                  "type bool")
 
             self.update = kwargs['update_qiita']
+
+
+class PrepNuQCWorkflow(StandardMetagenomicWorkflow):
+    def __init__(self, **kwargs):
+        qclient = kwargs['qclient']
+        job_id = kwargs['job_id']
+        parameters = kwargs['parameters']
+        out_dir = kwargs['out_dir']
+        config_fp = kwargs['config_fp']
+        status_line = kwargs['status_line']
+
+        out_path = partial(join, out_dir)
+        self.final_results_path = out_path('final_results')
+        makedirs(self.final_results_path, exist_ok=True)
+
+        pid = parameters.pop('prep_id')
+
+        prep_info = qclient.get(f'/qiita_db/prep_template/{pid}/')
+        dt = prep_info['data_type']
+        sid = prep_info['study']
+        if dt not in {'Metagenomic', 'Metatranscriptomic'}:
+            raise WorkflowError(f'Prep {pid} has a not valid data type: {dt}')
+        aid = prep_info['artifact']
+        if not str(aid).isnumeric():
+            raise WorkflowError(f'Prep {pid} has a not valid artifact: {aid}')
+
+        files, pt = qclient.artifact_and_preparation_files(aid)
+        html_summary = qclient.get_artifact_html_summary(aid)
+        if html_summary is None:
+            raise WorkflowError(f'Artifact {aid} doesnot have a summary, '
+                                'please generate one.')
+        df_summary = pd.read_html(html_summary)[0]
+        pt.set_index('sample_name', inplace=True)
+
+        project_name = f'qiita-{pid}-{aid}_{sid}'
+        run_identifier = '250225_LH00444_0301_B22N7T2LT4'
+
+        sheet = MetagenomicSampleSheetv90()
+        sheet.Header['IEMFileVersion'] = '4'
+        sheet.Header['Date'] = datetime.today().strftime('%m/%d/%y')
+        sheet.Header['Workflow'] = 'GenerateFASTQ'
+        sheet.Header['Application'] = 'FASTQ Only'
+        sheet.Header['Assay'] = prep_info['data_type']
+        sheet.Header['Description'] = f'prep_NuQCJob - {pid}'
+        sheet.Header['Chemistry'] = 'Default'
+        sheet.Header['SheetType'] = 'standard_metag'
+        sheet.Header['SheetVersion'] = '90'
+        sheet.Header['Investigator Name'] = 'Qiita'
+        sheet.Header['Experiment Name'] = project_name
+
+        sheet.Bioinformatics = pd.DataFrame(
+            columns=['Sample_Project', 'ForwardAdapter', 'ReverseAdapter',
+                     'library_construction_protocol',
+                     'experiment_design_description',
+                     'PolyGTrimming', 'HumanFiltering', 'QiitaID'],
```

Comment:
`'PolyGTrimming'` is not one of the bioinformatics fields that any of the existing sample sheet types (including v90) know about ... this is what we get for not enforcing validation.  Can you educate me on what this field is for and what sort(s) of sample sheets use it?

Reply:
Interesting, I guess I used an old version of a sample-sheet when I wrote this. Looking in my emails, I have an email saying that in a new sample-sheet I should  """Remove one column, it’s not used anymore: PolyGTrimming""". I can remove it.